crates/bevy_reflect/src/serde/de.rs

```diff
@@ -273,7 +274,7 @@ impl<'a, 'de> Visitor<'de> for UntypedReflectDeserializerVisitor<'a> {
         A: MapAccess<'de>,
     {
         let type_name = map
-            .next_key::<String>()?
+            .next_key::<Cow<'de, str>>()?
```

Comment:
AFAIK `Deserialize` for `Cow` is implemented to always deserialize the owned variant (thus `String` in our case), see https://github.com/serde-rs/serde/issues/1852
The issue also contains an example for how to make this work, although I think this is simplier:

```rust
#[derive(Deserialize)]
#[serde(transparent)]
struct BorrowableCowStr<'a>(
    #[serde(borrow)]
    Cow<'a, str>,
);
```

Reply:
Oh interesting. Maybe I'll give this a shot then? @james7132 any thoughts?

Reply:
That's unfortunate we have to do that, but if that's what it takes to avoid the allocation, that's OK.

Reply:
To be clear, it only avoids the allocation when deserializing a `&'de str` is possible.

From a quick look I think it would be possible to remove the allocation even when a shorter-lived `&str` is given, but that probably requires using `next_key_seed` and a custom `DeserializeSeed` struct to directly deserialize a `&TypeRegistration` (that borrows from the registry ofc) instead of a `&str`.

Reply:
Yeah I think a follow up PR would be good if there's still room to optimize 🙂